lib/src/components/comms_button/comms_button.dart

```diff
@@ -0,0 +1,426 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// Enum for the type of comms button.
+enum ZetaCommsButtonType {
+  /// Green background, no border, white icon
+  positive,
+
+  /// Red background, no border, white icon
+  negative,
+
+  /// Light grey background, dark grey border, black icon
+  on,
+
+  /// Dark grey background, light grey border, white icon
+  off,
+
+  /// Transparent background, red border, red icon
+  warning,
+}
+
+/// Comms button component.
+/// This component is used to display a button for communication action. Answer, reject, mute, hold, speaker, etc.
+/// {@category Components}
+///
+/// Figma: https://www.figma.com/design/JesXQFLaPJLc1BdBM4sisI/%F0%9F%A6%93-ZDS---Components?node-id=20816-7765&node-type=canvas&t=nc1YR061CeZRr6IJ-0
+///
+/// Widgetbook: https://zeta-ds.web.app/flutter/widgetbook/index.html#/?path=components/comms-button
+class ZetaCommsButton extends StatefulWidget {
+  /// Constructs [ZetaCommsButton]
+  const ZetaCommsButton({
+    super.key,
+    this.label,
+    this.type = ZetaCommsButtonType.on,
+    this.size = ZetaWidgetSize.medium,
+    this.icon,
+    this.onToggle,
+    this.toggledIcon,
+    this.toggledLabel,
+    this.toggledType,
+    this.onPressed,
+    this.focusNode,
+    this.semanticLabel,
+  });
+
+  /// Constructs answer call [ZetaCommsButton]
+  const ZetaCommsButton.answer({
+    super.key,
+    this.label,
+    this.size = ZetaWidgetSize.medium,
+    this.onToggle,
+    this.toggledIcon,
```

Comment:
If a button isn't toggleable such as this one, you can set a lot of these parameters to null, e.g. `onToggle`, `toggledType`, `toggledIcon`, etc.